airflow/www/views.py

```diff
@@ -2407,19 +2407,21 @@ def extra_links(self):
             return response
 
         task = dag.get_task(task_id)
-        try:
-            url = task.get_extra_links(dttm, link_name)
-        except ValueError as err:
-            response = jsonify({'url': None, 'error': str(err)})
-            response.status_code = 404
-            return response
-        if url:
-            response = jsonify({'error': None, 'url': url})
-            response.status_code = 200
+        if link_names:
+            name_url_map = {}
+            for single_name in link_names:
+                try:
+                    url = task.get_extra_links(dttm, single_name)
+                    if url:
+                        name_url_map[single_name] = {'error': None, 'url': url}
+                    else:
+                        name_url_map[single_name] = {'error': 'No URL found for {dest}'.format(dest=single_name), 'url': None}
+                except (ValueError, Exception) as err:
+                    name_url_map[single_name] = {'error': str(err), 'url': None}
+            response = jsonify(name_url_map)
             return response
         else:
-            response = jsonify(
-                {'url': None, 'error': 'No URL found for {dest}'.format(dest=link_name)})
+            response = jsonify({'url': None, 'error': "no link names provided"})
             response.status_code = 404
             return response
```

Comment:
Nit: perhaps invert the `if extra_links:` test above to *fail early*, before doing the heavy work of fetching the DAG and task objects?

So the structure becomes:

- Parse request args
- Check for extra_links; if missing -> 404
- Get DAG; if not found or task_id is not in the DAG-> 404
- Fetch the links and return 200 response.

Not having to run SQL in such a case is always a win.